sections/data_release_production_pipelines.tex

```diff
@@ -71,26 +75,332 @@ \section{Data Release Production}
+
+% Redefine the BootstrapImChar macro, since we'll refer back to those
+% sections here.
+\newcommand{\hr}[1]{\hyperref[sec:drpBootstrapImChar_#1]{#1}}
+
+\begin{lstlisting}
+def RefineImChar(%\hr{calexp}%, %\hr{sources}%, %\hr{reference}%):
+    matches = %\hr{MatchNonBlind}%(sources, reference)
+    %\hr{SelectStars}%(matches, exposures)
+    calexp.psf = %\hr{FitPSF}%(matches, sources, calexp.{mi,wcs})
+    parallel for ccd in ALL_SENSORS:
+        calexp[ccd].mi = %\hr{SubtractStars}%(calexp[ccd].{mi,psf}, sources[ccd])
+    calexp.{mi,background} = %\hr{SubtractBackground}%(calexp.mi)
+    parallel for ccd in ALL_SENSORS:
+        sources[ccd] = %\hr{DetectSources}%(calexp.{mi,psf})
+        calexp[ccd].mi, sources[ccd] =
+            %\hr{ReinsertStars}%(calexp[ccd].{mi,psf}, sources[ccd])
+        sources[ccd] = %\hr{DeblendSources}%(sources[ccd], calexp.{mi,psf})
+        sources[ccd] = %\hr{MeasureSources}%(sources[ccd], calexp.{mi,psf})
+    calexp.psf.apcorr = %\hr{FitApCorr}%(matches, sources)
+    parallel for ccd in ALL_SENSORS:
+        sources[ccd] = %\hr{ApplyApCorr}%(sources[ccd], calexp.psf)
+    return calexp, sources
+\end{lstlisting}
+
+This is essentially just another iteration of the loop in in \hyperref[sec:drpBootstrapImChar]{BootstrapImChar}, without the WCS-fitting or artifact-handling stages.  We assume that we continue to process the wavefront sensors here (because we will use them in the \hr{FitPSF} step), but it may be that previous processing may be sufficient.
+
+Note that RefineImChar does not update the CalExp's WCS, PhotoCalib, Image, or Variance (and its Mask is only updated to indicate new detections).
+
+
+% Undeclare the local hyperref macro
+\let\hr\undefined
+
 \subsubsection{FinalImChar}
 \label{sec:drpFinalImChar}
+
+FinalImChar is responsible for producting the final PSF models and source measurements.  While similar to \hyperref[sec:drpRefineImChar]{RefineImChar}, it is run after the \hyperref[sec:drpWarpAndPsfMatch]{WarpAndPsfMatch} and \hyperref[sec:drpBackgroundMatchAndReject]{BackgroundMatchAndReject} pipelines, which provide it with the final background model and and updated mask.
+
+The steps in FinalImChar are identical to those in \hyperref[sec:drpRefineImChar]{RefineImChar}, with just a few exceptions:
+
+\begin{itemize}
+\item The background is not re-estimated and subtracted.
+\item The suite of plugin run by \hyperref[sec:acSingleVisitMeasurement]{Single Visit Measurement} is expanded to included all algorithms indicated in the first column of Figure~\ref{fig:measurement-matrix}.  This should provide all measurements in the DPDD Source table description.
```

Comment:
I know we decided to go easy on the macros, but I think it'd be quite helpful to have macro that provided both linked text and a section number, while fitting into a naming scheme for sections that would let the label to reference be determined from the link text.  That's a level of standardization I clearly don't have time to do now (and I think it goes with coming up with CamelCase or otherwise code-like names for everything, which no one else seems to be doing though I've heard some enthusiastic noises).